src/main/java/it/gov/pagopa/payment/repository/TransactionInProgressRepositoryExtImpl.java

```diff
@@ -217,4 +220,35 @@ public List<TransactionInProgress> findByFilter(Criteria criteria, Pageable page
     public long getCount(Criteria criteria) {
         return mongoTemplate.count(Query.query(criteria), TransactionInProgress.class);
     }
+
+    @Override
+    public TransactionInProgress findCancelExpiredTransaction() {
+        return findExpiredTransaction(cancelExpirationMinutes, List.of(SyncTrxStatus.AUTHORIZED));
+    }
+
+    @Override
+    public TransactionInProgress findAuthorizationExpiredTransaction() {
+        return findExpiredTransaction(authorizationExpirationMinutes, List.of(SyncTrxStatus.IDENTIFIED, SyncTrxStatus.CREATED));
+    }
+
+    private TransactionInProgress findExpiredTransaction(long expirationMinutes, List<SyncTrxStatus> statusList) {
+        LocalDateTime now = LocalDateTime.now();
+
+        Query query = Query.query(
+                Criteria.where(Fields.trxChargeDate).lt(now.minusMinutes(expirationMinutes)).andOperator(
+                        Criteria.where(Fields.status).in(statusList)
+                                .orOperator(
+                                        Criteria.where(Fields.elaborationDateTime).is(null),
+                                        Criteria.where(Fields.elaborationDateTime).lt(now.minusSeconds(trxThrottlingSeconds))
+                                )
+                )
+        );
+
+        return mongoTemplate.findAndModify(query,
+                new Update()
+                        .currentDate(Fields.elaborationDateTime),
+                        // TODO update other fields?
```

Comment:
when throtthling we are not interested to change anything else, we can remove this TODO